linkerd/stack/src/fallback.rs

```diff
@@ -21,13 +23,21 @@ pub struct Fallback<I, F, P = fn(&Error) -> bool> {
     predicate: P,
 }
 
-pub enum MakeFuture<A, B, P> {
+#[pin_project]
+pub struct MakeFuture<A, B, P> {
+    #[pin]
+    state: State<A, B, P>,
```

Comment:
Any reason for adding the `State` enum here instead of keeping the `MakeFuture` as one?

Reply:
Yes, it's so that we can repeatedly pin project the state on multiple iterations of a loop even if we use [`Pin::set`](https://doc.rust-lang.org/std/pin/struct.Pin.html#method.set) to change the value between iterations. Tower does this in a few places as well.